contracts/RelayHub.sol

```diff
@@ -135,19 +142,24 @@ contract RelayHub is IRelayHub {
             IPaymaster(relayRequest.relayData.paymaster).getGasAndDataLimits{gas:50000}();
         require(msg.data.length <= gasAndDataLimits.calldataSizeLimit, "msg.data exceeded limit" );
         uint256 dataGasCost = calldataGasCost(msg.data.length);
+        uint256 txDataCost = externalGasLimit - initialGasLeft - relayCallDataOverhead;
+        uint256 txDataCostPerByte = txDataCost/msg.data.length;
+        require(txDataCostPerByte >= G_ZERO && txDataCostPerByte <= G_NONZERO, "invalid externalGasLimit");
```

Comment:
I am not sure this check is that useful. It requires adding 3 constants to the RelayHub for the benefit of setting very wide boundaries on the `externalGasLimit` which is still penalizeable if lied about. 

With false `externalGasLimit` the `actualCharge` can become higher than `maxPossibleGas` with or without this check.

Let's discuss this check further.

Reply:
Assuming we can't think of a relayer attack based on providing too little gas, we can remove the G_ZERO test
the G_NONZERO is there to prevent the OG-429 (externalGasLimit attack)

Its true that OG-429 is not strictly part of this issue, and the check could be a separate